akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala

```diff
@@ -59,34 +61,67 @@ object OverflowStrategy {
    * If the buffer is full when a new element arrives, drops the oldest element from the buffer to make space for
    * the new element.
    */
-  def dropHead: OverflowStrategy = DropHead
+  def dropHead: OverflowStrategy = DropHead(Logging.DebugLevel)
+
+  /**
+   * If the buffer is full when a new element arrives, drops the oldest element from the buffer to make space for
+   * the new element.
+   */
+  def dropHead(logLevel: LogLevel): OverflowStrategy = DropHead(logLevel)
```

Comment:
an alternative API could be
```
OverflowStrategy.dropHead.withLogLevel(Attributes.LogLevels.Warning)
```

Reply:
Yes, that will result in a simpler/less redundant API